### PR TITLE
feat: make the core site work offline

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Service workers are fun 👍

### CODE REVIEW

File: views/layouts/main.html [hunk 1]
```diff
@@ -13,6 +13,11 @@
     <div id="refinement-list"></div>
     {{> footer}}
     {{> anchor_links}}
+    <script>
+    if ('serviceWorker' in navigator) {
+      navigator.serviceWorker.register('/sw.js');
```
Comment (zeke) [inline views/layouts/main.html]: how about naming this file something like `service-worker.js` to make its purpose more obvious?

**Comment by zeke** [comment]
This is cool!

File: scripts/service-worker.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/* global importScripts, workbox */
+
+importScripts('https://storage.googleapis.com/workbox-cdn/releases/3.6.1/workbox-sw.js')
+importScripts('/scripts/precache.js')
+
+if (workbox) {
```
Comment (zeke) [inline scripts/service-worker.js]: Should this be `typeof workbox !== `undefined``?

**Comment by zeke** [comment]
Can you add a short blurb to the README about this feature, how it works, why it exists, etc?

File: scripts/service-worker.js (hunk 2 above)
Comment (MarshallOfSound) [reply scripts/service-worker.js]: Both work, will update for clarity

**Comment by MarshallOfSound** [comment]
@zeke note added 👍

**Pull request merged**